src/main/webapp/app/admin/statistics/statistics.component.ts

```diff
@@ -0,0 +1,165 @@
+    public barChartLegend = true;
+    public UserLoginChartData: ChartDataSets[] = [];
+    public SubmissionsChartData: ChartDataSets[] = [];
+
+    @ViewChild(BaseChartDirective) chart: BaseChartDirective;
+
+    constructor(private service: StatisticsService, private translateService: TranslateService) {}
+
+    async ngOnInit() {
+        await this.setBinWidth();
+        await this.createChart();
+    }
+
+    async ngOnChanges() {}
+
+    private async setBinWidth(): Promise<void> {
+        switch (this.span) {
+            case SpanType.DAY:
+                this.histogramData = Array(24).fill(0);
+                break;
+            case SpanType.WEEK:
+                this.histogramData = Array(7).fill(0);
+                break;
+            case SpanType.MONTH:
+                this.histogramData = Array(moment().daysInMonth()).fill(0);
+                break;
+            case SpanType.YEAR:
+                this.histogramData = Array(12).fill(0);
+                break;
+        }
+    }
+
+    async getSubmissions(): Promise<number[]> {
+        return new Promise<number[]>((resolve, reject) => {
+            this.service.getTotalSubmissions(this.span).subscribe((res: number[]) => {
+                if (res !== null) {
+                    resolve(res);
+                } else {
+                    reject('Submissions could not get fetched');
+                }
+            });
+        });
+    }
+
+    onTabChanged(span: string): void {
+        switch (span) {
+            case 'Day':
+                this.span = SpanType.DAY;
+                break;
+            case 'Week':
+                this.span = SpanType.WEEK;
+                break;
+            case 'Month':
+                this.span = SpanType.MONTH;
+                break;
+            case 'Year':
+                this.span = SpanType.YEAR;
+                break;
+        }
+    }
+
+    private getWeekdays(): string[] {
+        const days: string[] = [];
+
+        for (let i = 0; i < 7; i++) {
+            days.push(
+                moment()
+                    .subtract(6 - i, 'd')
+                    .format('dddd'),
+            );
+        }
+        return days;
+    }
+
+    private createLabels(): string[] {
+        let labels: string[] = [];
+        switch (this.span) {
+            case SpanType.DAY:
+                for (let i = 0; i < this.histogramData.length; i++) {
+                    labels[i] = `${i}:00,${i + 1}:00`;
+                }
+                break;
+            case SpanType.WEEK:
+                labels = this.getWeekdays();
+                break;
+            case SpanType.MONTH:
+                for (let i = 0; i < this.histogramData.length; i++) {
+                    labels[i] = i + 1 + '';
+                }
+                break;
+            case SpanType.YEAR:
+                labels = ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'];
```

Comment:
You should support both english and german

Reply:
Now does :)